daemons/fenced/fenced_remote.c

```diff
@@ -1592,26 +1611,33 @@ call_remote_stonith(remote_fencing_op_t * op, st_query_result_t * peer, int rc)
             g_source_remove(op->op_timer_one);
         }
 
-        if(stonith_watchdog_timeout_ms > 0 && device && pcmk__str_eq(device, "watchdog", pcmk__str_casei)) {
-            crm_notice("Waiting %lds for %s to self-fence (%s) for client %s "
-                       CRM_XS " id=%.8s", (stonith_watchdog_timeout_ms / 1000),
-                       op->target, op->action, op->client_name, op->id);
-            op->op_timer_one = g_timeout_add(stonith_watchdog_timeout_ms, remote_op_watchdog_done, op);
-
-            /* TODO check devices to verify watchdog will be in use */
-        } else if(stonith_watchdog_timeout_ms > 0
-                  && pcmk__str_eq(peer->host, op->target, pcmk__str_casei)
-                  && !pcmk__str_eq(op->action, "on", pcmk__str_casei)) {
-            crm_notice("Waiting %lds for %s to self-fence (%s) for client %s "
-                       CRM_XS " id=%.8s", (stonith_watchdog_timeout_ms / 1000),
-                       op->target, op->action, op->client_name, op->id);
-            op->op_timer_one = g_timeout_add(stonith_watchdog_timeout_ms, remote_op_watchdog_done, op);
-
-        } else {
+        if (!(stonith_watchdog_timeout_ms > 0 && (
+                (pcmk__str_eq(device, STONITH_WATCHDOG_ID,
+                                        pcmk__str_none)) ||
+                (pcmk__str_eq(peer->host, op->target, pcmk__str_casei)
+                    && !pcmk__str_eq(op->action, "on", pcmk__str_casei))) &&
+             check_watchdog_fencing_and_wait(op))) {
+
+            /* Some thoughts about self-fencing cases reaching this point:
+               - Actually check in check_watchdog_fencing_and_wait
+                 shouldn't fail if STONITH_WATCHDOG_ID is
+                 chosen as fencing-device and it being present implies
+                 watchdog-fencing is enabled anyway
+               - If watchdog-fencing is disabled either in general or for
+                 a specific target - detected in check_watchdog_fencing_and_wait -
+                 for some other kind of self-fencing we can't expect
+                 a success answer but timeout is fine if the node doesn't
+                 come back in between
+               - Delicate might be the case where we have watchdog-fencing
+                 enabled for a node but the watchdog-fencing-device isn't
+                 explicitly chosen for suicide. Local pe-execution in sbd
+                 may detect the node als unclean and lead to timely suicide.
```

Comment:
as

Reply:
that doesn't happen often when writing - hope it doesn't happen more often when I'm talking.
actually I thought I was quite good switching between languages - at least I have issues finding German words when in an English converstion.

Reply:
I thought it was just a regular typo, now I know one more German word :)

I've actually never noticed anything like that in your writing or talking. I'm always amazed at how fluent non-native speakers at RH are in English when I struggle to order lunch in German or Czech ...

Reply:
That is probably not really comparable.
In most countries children start learning English at a quite young age and you are much more exposed to it in everyday life.